Sources/PublicInterface/Client.swift

```diff
@@ -201,23 +204,29 @@ public class Client: WalletConnect {
 
             guard walletInfo.approved else {
                 // TODO: handle Error
-                delegate.client(self, didFailToConnect: response.url)
+                delegate?.client(self, didFailToConnect: response.url)
                 return
             }
 
             communicator.addSession(session)
-            delegate.client(self, didConnect: session)
+            delegate?.client(self, didConnect: session)
         } catch {
             // TODO: handle error
-            delegate.client(self, didFailToConnect: response.url)
+            delegate?.client(self, didFailToConnect: response.url)
         }
     }
 
     override func onTextReceive(_ text: String, from url: WCURL) {
         if let response = try? communicator.response(from: text, url: url) {
             log(response)
-            if let completion = responses.find(requestID: response.internalID) {
-                completion(response)
+            // Rainbow (at least) application send connect response with random id
```

Comment:
Then Rainbow violates JSON-RPC 2.0 spec (https://www.jsonrpc.org/specification#response_object)
Response on `wc_sessionRequest` should have same id as request.
@pedrouid do you know someone from Rainbow team to clarify on this? I would be against adding this change to the lib.

Reply:
I tested Rainbow wallet and now it works fine